tests/test_midi_stress.py

```diff
@@ -0,0 +1,121 @@
+from pathlib import Path
+import pytest
+import subprocess
+import time
+import json
+import platform
+import mido
+
+from test_midi import (
+    input_midi_file_name,
+    output_midi_file_name,
+    run_midi_test_file
+    )
+
+from usb_audio_test_utils import (
+    check_analyzer_output,
+    get_xtag_dut_and_harness,
+    AudioAnalyzerHarness,
+    wait_for_midi_ports,
+    find_xmos_midi_device,
+    XrunDut,
+    XsigInput,
+    XsigOutput,
+    xsig_completion_time_s
+)
+from conftest import list_configs, get_config_features
+
+def midi_common_uncollect(features, board, pytestconfig):
+    xtag_ids = get_xtag_dut_and_harness(pytestconfig, board)
+    # XTAGs not present
+    if not all(xtag_ids):
+        return True
+
+    # Until we fix Jenkins user permissions for MIDI on Mac https://xmosjira.atlassian.net/browse/UA-254
+    if platform.system() == "Darwin":
+        return True
+
+    if features["i2s_loopback"]:
+        return True
+    return False
+
+
+def midi_output_uncollect(pytestconfig, board, config):
+    features = get_config_features(board, config)
+    return any(
+        [not features["midi"], midi_common_uncollect(features, board, pytestconfig)]
+    )
+
+
+def midi_duration(level, partial):
+    if level == "weekend":
+        duration = 90 if partial else 1200
+    elif level == "nightly":
+        duration = 15 if partial else 180
+    else:
+        duration = 10
+    return duration
+
+
+@pytest.mark.uncollect_if(func=midi_output_uncollect)
+@pytest.mark.parametrize(["board", "config"], list_configs())
+def test_midi_loopback_stress(pytestconfig, board, config):
+    """
+    This test streams 8ch audio in/out at 192kHz in order to stress the system and then runs the 
+    standard test_midi.py test whilst doing so. Note we only check the xsig input for analog
+    (not the harness xscope output) because as soon as you stop either the harness or xsig the 
+    other will throw an error due to real-time checking.
+    """
+    features = get_config_features(board, config)
+
+    xsig_config = f'mc_midi_stress_8ch'
+    xsig_config_path = Path(__file__).parent / "xsig_configs" / f"{xsig_config}.json"
+
+    adapter_dut, adapter_harness = get_xtag_dut_and_harness(pytestconfig, board)
+    duration = midi_duration(pytestconfig.getoption("level"), features["partial"])
+    fail_str = ""
+
+    fs_audio = max(features["samp_freqs"]) # Highest rate for maximum stress
+    with XrunDut(adapter_dut, board, config) as dut:
+
+        # Ensure firmware is up and enumerated as MIDI
+        wait_for_midi_ports()
+
+        time_start = time.time()
+        with (
+            AudioAnalyzerHarness(
+                adapter_harness, xscope="io"
+            ) as harness,
+            # Due to in and out in xsig config this streams audio in both directions for max stress
+            XsigInput(fs_audio, duration, xsig_config_path, dut.dev_name, ident=f"analogue_input-{board}-{config}-{fs_audio}") as xsig_proc_in
```

Comment:
Replace "analogue_input" in the `ident` parameter to uniquely identify this test.